BrotliCompression task is not properly escaping the tool path

### Describe the bug
The BrotliCompression task launches a new process, but it is not properly escaping the path. So if you install the latest SDK into `%PROGRAMFILES%`, publish fails.

https://github.com/dotnet/aspnetcore/blob/088380b45cad29f48635867526b954d8c86b84f7/src/Components/WebAssembly/Sdk/src/BrotliCompress.cs#L57

Looking at the code, it also appears to not properly escape the file paths it passes into the program as arguments. So if your project was in a folder with a space in its name, it wouldn't work:

https://github.com/dotnet/aspnetcore/blob/088380b45cad29f48635867526b954d8c86b84f7/src/Components/WebAssembly/Sdk/src/BrotliCompress.cs#L92-L96

### To Reproduce
1. Install the latest 5.0 SDK - 5.0.100-rc.1.20420.7 - to `%PROGRAMFILES%`.
2. `dotnet new blazorwasm`
3. `dotnet publish -c Release`

### Exceptions (if any)
```
Compressing Blazor WebAssembly publish artifacts. This may take a while...
C:\Program Files\dotnet\sdk\5.0.100-rc.1.20420.7\Sdks\Microsoft.NET.Sdk.BlazorWebAssembly\targets\Microsoft.NET.Sdk.Blaz
orWebAssembly.Current.targets(525,5): error MSB6006: "C:\Program Files\dotnet" exited with code 1. [F:\DotNetTest\Blazor
5Test\Blazor5Test.csproj]
```

From the .binlog, I also see:

```
Could not execute because the specified command or file was not found.
```

### Further technical details
- ASP.NET Core version
- Include the output of `dotnet --info`

```
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.1.20420.7
 Commit:    3232fe7ac1

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.1.20420.7\

Host (useful for support):
  Version: 5.0.0-rc.1.20417.14
  Commit:  907f7da59b
```

- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version

cc @pranavkm

--- a/src/Components/WebAssembly/Sdk/src/BrotliCompress.cs
+++ b/src/Components/WebAssembly/Sdk/src/BrotliCompress.cs
@@ -54,7 +54,18 @@ namespace Microsoft.NET.Sdk.BlazorWebAssembly
             }
         }
 
-        protected override string GenerateCommandLineCommands() => ToolAssembly;
+        private static string Quote(string path)
+        {
+            if (string.IsNullOrEmpty(path) || (path[0] == '\"' && path[path.Length - 1] == '\"'))
+            {
+                // it's already quoted
+                return path;
+            }
+
+            return $"\"{path}\"";
+        }
+
+        protected override string GenerateCommandLineCommands() => Quote(ToolAssembly);
 
         protected override string GenerateResponseFileCommands()
         {

